src/instana/util/config.py

```diff
@@ -82,18 +82,26 @@ def parse_endpoints_of_service(
     """
     if service == "kafka" and isinstance(methods, list):
         for rule in methods:
-            for method, endpoint in itertools.product(
-                rule["methods"], rule["endpoints"]
-            ):
-                ignore_endpoints.append(
-                    f"{service.lower()}.{method.lower()}.{endpoint.lower()}"
-                )
+            ignore_endpoints.extend(parse_kafka_methods(rule))
     else:
         for method in methods:
             ignore_endpoints.append(f"{service.lower()}.{method.lower()}")
     return ignore_endpoints
 
 
+def parse_kafka_methods(rule: Union[str, Dict[str, any]]):
```

Comment:
I loved it!!! You are making the code cleaner ;-) 
However, you forgot to add the type annotation for the return of the function.